File: hello/src/main/scala/com/scala/clzAndObj/Student.scala
package com.scala.clzAndObj

class Student(name: String, age: Int) {
  def printInfo(): Unit = {
    println(this.name + " " + this.age + " " + Student.school)
  }
}

object Student {
  val school: String = "xmu"
  
  def main(args: Array[String]): Unit = {
    val zs = new Student("zs", 20)
    val ls = new Student("ls", 22)
    zs.printInfo()
    ls.printInfo()
  }
}

File: hello/src/main/java/com/java/clzAndObj/Student.java
package com.java.clzAndObj;

/**
 * {@code @description:}
 */
public class Student {
    private static final String school = "xmu";
    private String name;
    private Integer age;
    
    public Student(String name, Integer age) {
        this.name = name;
        this.age = age;
    }
    
    public static void main(String[] args) {
        Student zs = new Student("zs", 20);
        Student ls = new Student("ls", 22);
        zs.printInfo();
        ls.printInfo();
    }
    
    private void printInfo() {
        System.out.println(this.name + " " + this.age + " " + Student.school);
    }
}
